pkg/config/constants/constants.go

```diff
@@ -206,4 +206,6 @@ const (
 	AllTraffic = "all"
 	// NoTraffic indicates that no traffic should go through the intended waypoint.
 	NoTraffic = "none"
+
+	DisableV2AutoAllocationLabel = "networking.istio.io/disable-autoallocate-ip"
```

Comment:
Please change this to EnableAutoallocation - we can't ask users to opt-out all the 'route destination SE' or all HTTP-only SE ( that rely on port ) when enabling the auto alloc for few entries.

Also it is likely a better pattern to have the feature flag include the label name - user may have some label they 
use to select SE that are used as 'TCP frontend' and 'TCP backend' ( since OPA or other checks are required 
on SE to keep the security)

Reply:
John pointed out, and I am adding it now, that if resolution is `none` we should not auto-assign. So the ux would really be if I choose resolution!=none and don't assign an address and didn't opt out explicitly and the feature is enabled then I get addresses. Does that change affect your thinking on this?

Reply:
I don't think this should be opt-out - right now majority of users don't enable the auto-allocation and I don't think allocating IPs to all 'backend' SE and subtly changing the behavior is reasonable.

Resolution 'none' is not defined as 'autoallocate IP' - there are use cases for it that are documented. It was not defined like Service 'none'. 

I think it would be safer, more secure and simpler if user can explicitly indicate that they want a FE address manually assigned or if they want a controller to assign one. Even in Service - users have explicit 'first class' API
to specify if they want an LB or controller associated with the service, or ClusterIP or None ( but not the same none we have - ours is about resolution, theirs is about the IP allocated to the Service :-).

Mixing the 2 'none' would be extremely confusing and pretty guaranteed to break something on upgrade ( if admin enables this feature for some workloads - but some others are accidentally affected).

A more drastic approach ( which I am working on separately, for increasing the security posture and integration with external DNS)  is to add a namespace  where SE will get allocated addresses ( with RBAC ). The 'opt-in' labels have a second purpose - to allow users to set OPA and other policies which will be required to avoid hijacking DNS ( they are required now too - but we don't document it, left for users to figure out)

Reply:
The proposal includes a meshconfig setting to make it opt in vs opt out: https://docs.google.com/document/d/1dHyZPKQXxs022yXQgRpYN_cD_ZFbzxHJRHCX_rmG5vk/edit#heading=h.ogti5k7jbgpa.

Both are critical use cases, IMO.

We  must have the _option_ to make all SE auto-allocate to match existing use cases.

Probably makes sense to have:
* Default if nothing is configured: opt-in per SE
* If you enable some mode: all SE are allocated, can opt-out per SE

On top of that there is a temporary feature flag to disable the entire thing; this is not a user facing API or config knob, just an escape hatch if the controller is buggy.